### PR TITLE
xplat commands for Server Management Tools

### PR DESCRIPTION
**Pull request by fearthecowboy**
The content to be added to Changelog is as follows:
* Adding commands for Server Management Tools service

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@fearthecowboy__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (garretts). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

File: lib/commands/arm/servermanagement/servermanagement._js [hunk 1]
```diff
@@ -0,0 +1,481 @@
...
+    .description($('Commands to manage Azure Server Management Tools Sessions'));
+  var powershell = sm.category('powershell')
+    .description($('Commands to invoke powershelll commands on a Azure Server Management Tools node'));
+
+  gateway.command('create [resource-group] [location] [gateway-name]')
+    .description($('Create a server management gateway'))
+    .usage('[options] <resource-group> <location> <gateway-name>')
+    .option(help.resourceGroup.option, $(help.resourceGroup.description))
+    .option(help.location.option, $(help.location.description))
+    .option(help.gatewayName.option, $(help.gatewayName.description))
```
Comment (amarzavery) [inline lib/commands/arm/servermanagement/servermanagement._js]: -n --name for the artifact in this case gateway that is being operated upon. You can use --gateway-name in other commands

Comment (amarzavery) [inline lib/commands/arm/servermanagement/servermanagement._js]: same convention everywhere

File: lib/commands/arm/servermanagement/servermanagement._js [hunk 2]
```diff
@@ -0,0 +1,481 @@
...
+  row.cell($('Resource Group'), resourceGroup);
+  row.cell($('Location'), gateway.location);
+  row.cell($('Auto Upgrade'), gateway.autoUpgrade);
+}
+
+function formatGatewayDetailed(row, gateway) {
+  for (var each in gateway.instances) {
+    row.cell($('Gateway'), gateway.name);
+    row.cell($('Instance'), gateway.instances[each].name);
+    row.cell($('Mem Avail (MB)'), gateway.instances[each].availableMemoryMByte);
```
Comment (amarzavery) [inline lib/commands/arm/servermanagement/servermanagement._js]: Please make sure that all these properties will always be present in every item in the array. If that does not happen then javascript will throw an error Cannot access availableMemoryMByte of undefined. It is a good practice to check before using something. If this is always going to be there then it is ok. I have taken this as an example. It is a general comment.

File: lib/commands/arm/servermanagement/servermanagement._js (hunk 1 above)
Comment (fearthecowboy) [inline lib/commands/arm/servermanagement/servermanagement._js]: I thought about doing it that way; It's kind of a pain for users tho, isn't it?

File: lib/commands/arm/servermanagement/servermanagement._js (hunk 2 above)
Comment (fearthecowboy) [inline lib/commands/arm/servermanagement/servermanagement._js]: In these cases, they are definitely always there.

File: lib/commands/arm/servermanagement/servermanagement._js (hunk 1 above)
Comment (amarzavery) [inline lib/commands/arm/servermanagement/servermanagement._js]: yup. Most of them get used to the convention that using -n or --name is guaranteed to work. With this they have to again learn new things.

File: test/commands/arm/servermanagement/arm.servermanagement-crud-tests.js [hunk 3]
```diff
@@ -0,0 +1,402 @@
...
+
+  describe('servermanagement', function() {
+    var suite;
+    var testVault;
+
+    before(function(done) {
+      suite = new CLITest(this, testprefix, requiredEnvironment);
+
+      // for interactive debugging
+      suite._execute = suite.execute;
```
Comment (amarzavery) [inline test/commands/arm/servermanagement/arm.servermanagement-crud-tests.js]: you dont need this. The time stamped Test logs for every run are present in the tests/outputput folder which gives you all the info

Comment (fearthecowboy) [inline test/commands/arm/servermanagement/arm.servermanagement-crud-tests.js]: Ah, yeah... I forgot to pull that out. I did that during development to track what I was running interactively which helped when `done()` never got called.

Comment (amarzavery) [inline test/commands/arm/servermanagement/arm.servermanagement-crud-tests.js]: Ah I see. However, do look at the logs. It has this nice ---start---- and ----end--- indicator for each test and shows the set of commands that ran in that test. It also lets you know whether that test passed or failed.

**Comment by fearthecowboy** [comment]
Hmm. It's still failing to find the `azure-arm-servermanagement` module

It runs just fine locally.

**Comment by amarzavery** [comment]
@fearthecowboy - Please update the PR description with the content that needs to go in the changelog at the end of the release

**Comment by azurecla** [comment]
Hi __@fearthecowboy__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (garretts). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

**Pull request merged**